### PR TITLE
Issue 38 msg clean up

### PR DESCRIPTION
**Pull request by tko-dev**
## smarc2 Changes
[Issue Ticket](https://github.com/smarc-project/smarc2/issues/38)
Refactoring the following messages from `sam_msg` -> `smarc_msg`:
- [x] `Leak`
- [x] `PercentStamped`

Goal was to remove any unused messages in smarc2 thereby clearing out old legacy messages. Removing the following unused messages.
- `sam_msg`
	- [x] `ArrayCommand`
	- [x] `BallastAngles`
	- [x] `CTP`
	- [x] `CircuitStatus*`
	- [x] `Command`
	- [x] `Consumed*`
	- [x] `UavcanNode*`
	- [x] `LightCommandArray*`
	- [x] `UavcanNode*`
- `smarc_msg`
	- [x] `CTD`
	- [x] `CommsMessage`
	- [x] `ControllerStatus`
	- [x] `DualThruster*`
	- [x] `ExecutionStatus`
	- [x] `LatLon*`
	- [x] `StringPair`
	- [x] `SMTask`
	- [x] `StringArray`
	- [x] `SensorStatus`

Large amount of ROS 1 Scripts were removed as they are no longer relevant. See file changes for more details

## SMARCUnityAssets Updates
MR changes require refactoring of `SMARCUnityAssets` which is captured on the following [MR](https://github.com/smarc-project/SMARCUnityAssets/pull/83)

### CODE REVIEW

File: messages/sam_msgs/CMakeLists.txt [hunk 1]
```diff
@@ -14,28 +14,11 @@ find_package(sensor_msgs REQUIRED)
...
-  "msg/ConsumedChargeArray.msg"
-  "msg/ConsumedChargeFeedback.msg"
-  "msg/ConsumedCharge.msg"
-  "msg/CTP.msg"
-  "msg/Leak.msg"
-  "msg/LightCommandArray.msg"
-  "msg/LightCommand.msg"
-  "msg/PercentStamped.msg"
+  "msg/JoyButtons.msg"
+  "msg/Links.msg"
```
Comment (tko-dev) [inline messages/sam_msgs/CMakeLists.txt]: Just moved messages for better readability (not actually added)

File: examples/sam_basic_controllers/src/pid_utils/pid_actuator_trim.cpp
Comment (tko-dev) [inline examples/sam_basic_controllers/src/pid_utils/pid_actuator_trim.cpp:1]: Updated basic controller C++ imports to ensure controller keeps working with all refactoring

File: examples/sam_basic_controllers/package.xml [hunk 2]
```diff
@@ -16,6 +16,7 @@
   since the entire parameter system has changed! -->
   <!-- <depend>dynamic_reconfigure</depend> -->
   <depend>sam_msgs</depend>
+  <depend>smarc_msgs</depend>
   <depend>std_msgs</depend>
```
Comment (tko-dev) [inline examples/sam_basic_controllers/package.xml:20]: Need to remove all but std_msgs

File: messages/sam_msgs/CMakeLists.txt [hunk 3]
```diff
@@ -14,28 +14,11 @@ find_package(sensor_msgs REQUIRED)
...
   "msg/ThrusterAngles.msg"
   "msg/ThrusterRPMs.msg"
-  "msg/UavcanNodeStatus.msg"
-  "msg/UavcanNodeStatusNamedArray.msg"
-  "msg/UavcanNodeStatusNamed.msg"
   "msg/Topics.msg"
-  "msg/Links.msg"
-  "msg/JoyButtons.msg"
   "srv/UavcanUpdateBattery.srv"
   "action/SystemsCheck.action"
```
Comment (KKalem) [inline messages/sam_msgs/CMakeLists.txt]: nuke

File: examples/sam_basic_controllers/package.xml [hunk 4]
```diff
@@ -16,6 +16,7 @@
   since the entire parameter system has changed! -->
   <!-- <depend>dynamic_reconfigure</depend> -->
   <depend>sam_msgs</depend>
+  <depend>smarc_msgs</depend>
```
Comment (KKalem) [inline examples/sam_basic_controllers/package.xml:19]: nuke

File: messages/smarc_msgs/CMakeLists.txt [hunk 5]
```diff
@@ -22,23 +22,18 @@ rosidl_generate_interfaces(${PROJECT_NAME}
   "srv/AddTasks.srv"
   "srv/NEDENURotation.srv"
   "srv/LatLonToUTMOdometry.srv"
```
Comment (KKalem) [inline messages/smarc_msgs/CMakeLists.txt]: nuke

File: messages/smarc_msgs/CMakeLists.txt [hunk 6]
```diff
@@ -22,23 +22,18 @@ rosidl_generate_interfaces(${PROJECT_NAME}
...
-  "msg/CommsMessage.msg"
   "msg/ControllerStatus.msg"
```
Comment (KKalem) [inline messages/smarc_msgs/CMakeLists.txt]: double check

File: messages/smarc_msgs/CMakeLists.txt [hunk 7]
```diff
@@ -22,23 +22,18 @@ rosidl_generate_interfaces(${PROJECT_NAME}
...
   "msg/CTD.msg"
```
Comment (KKalem) [inline messages/smarc_msgs/CMakeLists.txt]: nuke

File: messages/smarc_msgs/CMakeLists.txt [hunk 8]
```diff
@@ -22,23 +22,18 @@ rosidl_generate_interfaces(${PROJECT_NAME}
...
-  "msg/DualThrusterFeedback.msg"
-  "msg/DualThrusterRPM.msg"
   "msg/DVLBeam.msg"
   "msg/DVL.msg"
-  "msg/ExecutionStatus.msg"
   "msg/FloatStamped.msg"
   "msg/LatLonOdometry.msg"
```
Comment (KKalem) [inline messages/smarc_msgs/CMakeLists.txt]: nuke

File: messages/smarc_msgs/CMakeLists.txt [hunk 9]
```diff
@@ -22,23 +22,18 @@ rosidl_generate_interfaces(${PROJECT_NAME}
...
   "msg/LatLonStamped.msg"
```
Comment (KKalem) [inline messages/smarc_msgs/CMakeLists.txt]: nuke

File: messages/smarc_msgs/CMakeLists.txt [hunk 10]
```diff
@@ -22,23 +22,18 @@ rosidl_generate_interfaces(${PROJECT_NAME}
...
   "msg/Leak.msg"
-  "msg/SensorStatus.msg"
+  "msg/PercentStamped.msg"
   "msg/Sidescan.msg"
   "msg/SMTask.msg"
```
Comment (KKalem) [inline messages/smarc_msgs/CMakeLists.txt]: nuke

File: messages/smarc_msgs/CMakeLists.txt [hunk 11]
```diff
@@ -22,23 +22,18 @@ rosidl_generate_interfaces(${PROJECT_NAME}
...
   "msg/StringArray.msg"
```
Comment (KKalem) [inline messages/smarc_msgs/CMakeLists.txt]: nuke

File: simulation/SMARCUnityAssets
Comment (KKalem) [inline simulation/SMARCUnityAssets:1]: plz

**Review by KKalem — CHANGES_REQUESTED** [review]

File: messages/smarc_msgs/CMakeLists.txt [hunk 12]
```diff
@@ -17,28 +17,14 @@ find_package(std_msgs REQUIRED)
...
   "msg/DVL.msg"
-  "msg/ExecutionStatus.msg"
   "msg/FloatStamped.msg"
-  "msg/LatLonOdometry.msg"
-  "msg/LatLonStamped.msg"
   "msg/Leak.msg"
-  "msg/SensorStatus.msg"
+  "msg/PercentStamped.msg"
   "msg/Sidescan.msg"
   "msg/SMTask.msg"
```
Comment (KKalem) [inline messages/smarc_msgs/CMakeLists.txt]: Nuke this and this script: https://github.com/smarc-project/smarc2/blob/humble/messages/smarc_msgs/src/smarc_msgs/sm_task_utils.py

File: messages/smarc_msgs/CMakeLists.txt [hunk 13]
```diff
@@ -17,28 +17,14 @@ find_package(std_msgs REQUIRED)
...
   "msg/StringArray.msg"
```
Comment (KKalem) [inline messages/smarc_msgs/CMakeLists.txt]: this too

**Review by KKalem — CHANGES_REQUESTED** [review]
last small bits

**Comment by KKalem** [comment]
Also nuking https://github.com/smarc-project/smarc2/blob/758d90a26c60b7d167248274a26d60c44da35917/gui/smarc_nodered/src/waraps_bridge.py#L5 while we're at it.

**Comment by tko-dev** [comment]
Sim is happy, hit all the waypoints with a simple small test

**Review by KKalem — APPROVED** [review]

**Pull request merged**